addons/controls/src/ControlsPanel.tsx

```diff
@@ -17,10 +17,17 @@ export const ControlsPanel: FC = () => {
     PARAM_KEY,
     {}
   );
-  const hasControls = Object.values(rows).filter((argType) => !!argType?.control).length > 0;
+
+  const hasControls = useMemo(() => Object.values(rows).some((arg) => arg?.control), [rows]);
```

Comment:
TBH this feels like premature optimization too. How many rows are going to be necessary for this to make any difference?